interfaces/cython/cantera/base.pyx

```diff
@@ -111,8 +109,32 @@ cdef class _SolutionBase:
         elif source:
             root = AnyMapFromYamlString(stringify(source))
 
-        phaseNode = root[stringify("phases")].getMapWhere(stringify("name"),
-                                                          stringify(name))
+        try:
+            phaseNode = root[stringify("phases")].getMapWhere(
+                stringify("name"), stringify(name))
+        except CanteraError as err:
```

Comment:
I'm not really a fan of this approach either. One reason is that any errors that are generated during parsing of the file I think would have a double layer of stack trace printed:
```python
>>> try:
...     raise ValueError("Error")
... except ValueError as err:
...     raise err
...
Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
  File "<stdin>", line 2, in <module>
ValueError: Error
```
(although you can get around that by
```python
>>> try:
...     raise ValueError("Error")
... except ValueError:
...     raise
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ValueError: Error
```
)
I'd really prefer if there were a way to check that the file is being loaded from the Cantera data location, since from my perspective, that's the only one that we have control over and therefore the only one we should care about. The `AnyMap` has a `__file__` field, I believe, can that be checked for one of the known Cantera locations?

Reply:
Since this will only stay in temporarily for 2.5, I'd prefer to use the default mechanism (while adopting your suggestion).

Reply:
> Since this will only stay in temporarily for 2.5, I'd prefer to use the default mechanism

I'm not sure what this means?

Reply:
Stack trace is fixed.

Reply:
Default mechanism means that I'll let let:
```
phaseNode = root[stringify("phases")].getMapWhere(
                 stringify("name"), stringify(name))
```
check whether the phase is actually defined. I.e. I referred to `getMapWhere`

Reply:
Default mechanism means that I'll let let:
```
phaseNode = root[stringify("phases")].getMapWhere(
                 stringify("name"), stringify(name))
```
check whether the phase is actually defined. I.e. I referred to `getMapWhere`

Reply:
Default mechanism means that I'll let let:
```
phaseNode = root[stringify("phases")].getMapWhere(
                 stringify("name"), stringify(name))
```
check whether the phase is actually defined. I.e. I referred to `getMapWhere`

Reply:
Stack is fixed, btw.

Reply:
stack is fixed btw.

Reply:
> 'd really prefer if there were a way to check that the file is being loaded from the Cantera data location, since from my perspective, that's the only one that we have control over and therefore the only one we should care about.

I'm not sure I agree: the (temporary) check should warn users that they are trying to load `gri30_mix` or `gri30_multi` from a `YAML` file that doesn't contain it. If they traded a *new* `gri30.yaml` from any other location than the official one, the same rationale would apply? After this revision, no exception is raised if the phase is actually defined. After 2.5, users get an error if they try to load `gri30_mix` from the official `gri30.yaml` and the rest will still work as intended.

Reply:
> If they traded a new gri30.yaml from any other location than the official one, the same rationale would apply?

OK, I can see this point. If the `gri30_mix` or `gri30_multi` phases are present in a YAML file they will be used. The only time this warning is shown is if the user-specified phase name is not in the YAML file and they have passed one of those two options. This seems reasonable.

Reply:
However, I think the warning message needs to be improved in this case. If a user naively converts the CK version of GRI-3.0 to YAML and uses it with an example that shows the use of these phases, the error message might not make much sense to them. I'll leave a suggestion in the error message.

Reply:
Hm. There won't be any difference in terms what the code does, as the `gri30.yaml` file will be (presumably) identical to the 'official' version. But I do see your point.

Reply:
Resolved.